packages/in_app_purchase/in_app_purchase_android/lib/src/types/in_app_purchase_exception.dart

```diff
@@ -0,0 +1,23 @@
+/// Thrown to indicate that an action failed while interacting with the
+/// in_app_purchase plugin.
+class InAppPurchaseException implements Exception {
```

Comment:
Is this android only? Does it make sense to be in the platform_interface?

Reply:
At the moment it is only used in the Android package. On iOS we don't need this specific exception. However I can imagine the exception to be useful on other platforms in the future. 

So I wasn't sure to add it now to the platform_interface or later if we really need it on multiple platform (by that time we might have forgotten that there is a similar class in the Android implementation).

Reply:
Yes, I think it belongs to the platform_interface

Reply:
I created a separate PR (#3852) which adds the `InAppPurchaseException` to the platform_interface. Once that is merged I will make sure it gets removed from this PR.